corehq/apps/reports/standard/deployments.py

```diff
@@ -272,22 +272,29 @@ def get_bulk_ancestors(self, location_ids):
         :param kwargs: extra parameters
         :return: dict
         """
-        where = Q(domain=self.domain, location_id__in=location_ids)
-        location_ancestors = SQLLocation.objects.get_ancestors(where)
-        location_by_id = {location.location_id: location for location in location_ancestors}
-        location_by_pk = {location.id: location for location in location_ancestors}
-
-        grouped_location = {}
-        for location_id in location_ids:
-
-            location_parents = []
-            current_location = location_by_id[location_id].id if location_id in location_by_id else None
+        # list of ancestors of a location
+        ancestors_by_location = {}
+
+        def get_ancestors(loc):
+            if loc.pk in ancestors_by_location:
+                return ancestors_by_location[loc.pk]
+            all_ancestors = []
+            if loc.parent:
+                all_ancestors.extend(loc.parent)
+                if loc.parent_id in ancestors_by_location:
+                    parent_ancestors = ancestors_by_location[loc.parent_id]
+                else:
+                    parent_ancestors = get_ancestors(loc.parent)
+                all_ancestors.extend(parent_ancestors)
```

Comment:
```suggestion
                all_ancestors.extend(get_ancestors(loc.parent))
```
Since the first part of `get_ancestors` already checks for an existing parent, I think you can just do this